src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets

```diff
@@ -263,13 +263,15 @@
 
     <Exec Command="whoami" Condition="'$(OSEnvironment)'!='Windows_NT' AND '$(VersionUserName)'==''" StandardOutputImportance="Low" IgnoreExitCode="true" IgnoreStandardErrorWarningFormat="true" ConsoleToMSBuild="true">
       <Output TaskParameter="ConsoleOutput" PropertyName="VersionUserName" />
+      <Output TaskParameter="ExitCode" PropertyName="UserNameExitCode" />
     </Exec>
     <Exec Command="hostname" Condition="'$(OSEnvironment)'!='Windows_NT' AND '$(VersionHostName)'==''" StandardOutputImportance="Low" IgnoreExitCode="true" IgnoreStandardErrorWarningFormat="true" ConsoleToMSBuild="true">
       <Output TaskParameter="ConsoleOutput" PropertyName="VersionHostName" />
+      <Output TaskParameter="ExitCode" PropertyName="HostNameExitCode" />
     </Exec>
 
     <PropertyGroup>
-      <BuiltByString Condition="'$(BuiltByString)'==''">%20built by: $(VersionUserName)-$(VersionHostName)</BuiltByString>
+      <BuiltByString Condition="'$(BuiltByString)'=='' AND '$(UserNameExitCode)'=='0' AND '$(HostNameExitCode)'=='0'">%20built by: $(VersionUserName)-$(VersionHostName)</BuiltByString>
```

Comment:
I think you are right, I didn't think of that. We can't check if VersionUserName/VersionHostName is empty, because if the command fails because whoami is not installed, then we will get VersionUserName="error: unkown command whoami" or something like that. One thing I can do though is to set the exit codes to 0 in case they are not defined yet one line above this. What  do you think about that?